pkg/transport/transport.go

```diff
@@ -65,9 +81,10 @@ func (transport *Transport) DialNode(ctx context.Context, node *pb.Node, opts ..
 		dialOption,
 		grpc.WithBlock(),
 		grpc.FailOnNonTempDialError(true),
+		grpc.WithUnaryInterceptor(InvokeTimeout{transport.requestTimeout}.Intercept),
 	}, opts...)
 
-	timedCtx, cancel := context.WithTimeout(ctx, connWaitTimeout)
+	timedCtx, cancel := context.WithTimeout(ctx, transport.requestTimeout)
```

Comment:
connect timeout and request timeout are different things

Reply:
okay so there should be different timeout values for each one? should they both be field values on `Transport`?

Reply:
yeah, but not sure whether they should have different values

Reply:
😬I guess I'll just set their default values as `20*time.Second` for now.

Reply:
I changed `connWaitTimeout` to `defaultDialTimeout` - does that seem better?